src/Microsoft.OData.Edm/Csdl/Parsing/SchemaJsonParser.cs

```diff
@@ -655,13 +655,20 @@ internal static CsdlKey ParseCsdlKey(string name, JsonElement keyArray, JsonPars
             {
                 if (v.ValueKind == JsonValueKind.Object)
                 {
-                    // TODO: ODL doesn't support the key referencing a property of a complex type as below.
                     // "$Key": [
                     //    {
                     //      "EntityInfoID": "Info/ID"
                     //    }
                     //  ]
-                    p.ReportError(EdmErrorCode.InvalidKeyValue, "It doesn't support the key object.");
+                    string pName = "";
+                    string pValue = "";

```

Comment:
Is there a variable by the name `propertyValue` in scope such that we cannot use that in place of `pValue`?